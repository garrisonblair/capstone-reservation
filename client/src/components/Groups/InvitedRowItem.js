import React, { Component } from 'react';
import { Button, List, Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';

class InvitedRowItem extends Component {
  handleDeletion = () => {
    const { deleteFunction, selectedInvitation } = this.props;
    deleteFunction(selectedInvitation.id);
  }

  render() {
    const { selectedInvitation, isAdmin } = this.props;
    console.log(isAdmin);
    return (
      <List.Item>
        <List.Content floated="left">
          <h3>
            <Icon name="envelope" />
            {selectedInvitation.invited_booker.user.username}
          </h3>
        </List.Content>
        <List.Content floated="right">
          {isAdmin === true ? <Button onClick={this.handleDeletion}>Remove</Button> : ''}
        </List.Content>
      </List.Item>
    );
  }
}

InvitedRowItem.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  selectedInvitation: PropTypes.object.isRequired,
  deleteFunction: PropTypes.func.isRequired,
  isAdmin: PropTypes.bool.isRequired,
};

export default InvitedRowItem;
