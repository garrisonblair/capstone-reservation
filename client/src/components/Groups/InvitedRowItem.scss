.ui.dimmer .ui.loader:after {
    border-color:  black transparent transparent
}
