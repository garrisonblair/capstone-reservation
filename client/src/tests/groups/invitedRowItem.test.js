/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import InvitedRowItem from '../../components/Groups/InvitedRowItem';

function setup() {
  const selectedInvitation = {
    id: 1,
    invited_booker: {
      username: 'myUsername',
    },
  };
  const deleteFunction = () => { };

  return { selectedInvitation, deleteFunction };
}

it('Shows row', () => {
  const { selectedInvitation, deleteFunction } = setup();
  const tree = renderer
    .create(
      <InvitedRowItem
        selectedInvitation={selectedInvitation}
        deleteFunction={deleteFunction}
        isAdmin={false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Shows row as admin', () => {
  const { selectedInvitation, deleteFunction } = setup();
  const tree = renderer
    .create(
      <InvitedRowItem
        selectedInvitation={selectedInvitation}
        deleteFunction={deleteFunction}
        isAdmin
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Shows row loading', () => {
  const { selectedInvitation, deleteFunction } = setup();
  const tree = renderer
    .create(
      <InvitedRowItem
        selectedInvitation={selectedInvitation}
        deleteFunction={deleteFunction}
        isAdmin
        isLoadingForTesting
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
