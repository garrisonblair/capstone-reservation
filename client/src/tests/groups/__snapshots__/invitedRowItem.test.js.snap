// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows row 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="left floated content"
  >
    <h3>
      <i
        aria-hidden="true"
        className="envelope icon"
      />
      myUsername
    </h3>
  </div>
  <div
    className="right floated content"
  >
    
  </div>
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui loader"
      />
    </div>
  </div>
</div>
`;

exports[`Shows row as admin 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="left floated content"
  >
    <h3>
      <i
        aria-hidden="true"
        className="envelope icon"
      />
      myUsername
    </h3>
  </div>
  <div
    className="right floated content"
  >
    <button
      className="ui button"
      onClick={[Function]}
      role="button"
    >
      Remove
    </button>
  </div>
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui loader"
      />
    </div>
  </div>
</div>
`;

exports[`Shows row loading 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="left floated content"
  >
    <h3>
      <i
        aria-hidden="true"
        className="envelope icon"
      />
      myUsername
    </h3>
  </div>
  <div
    className="right floated content"
  >
    
  </div>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui loader"
      />
    </div>
  </div>
</div>
`;
